pkg/ddc/thin/referencedataset/cm.go

```diff
@@ -0,0 +1,138 @@
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func createConfigMapForRefDataset(client client.Client, refDataset *datav1alpha1.Dataset, mountedRuntime base.RuntimeInfoInterface) error {
+	mountedRuntimeType := mountedRuntime.GetRuntimeType()
+	mountedRuntimeName := mountedRuntime.GetName()
+	mountedRuntimeNamespace := mountedRuntime.GetNamespace()
+
+	refNameSpace := refDataset.GetNamespace()
+
+	// add owner reference to ensure config map deleted when delete the dataset
+	ownerReference := metav1.OwnerReference{
+		APIVersion: refDataset.APIVersion,
+		Kind:       refDataset.Kind,
+		Name:       refDataset.Name,
+		UID:        refDataset.UID,
+	}
+
+	// copy the configmap to ref namespace.
+	// TODO: any other config resource like secret need copied ?
+
+	// Note: values configmap is not needed for fuse sidecar container.
+
+	// TODO: decoupling the switch-case, too fragile
+	switch mountedRuntimeType {
+	// TODO:  currently the dst configmap name is the same as src configmap name to avoid modify the fuse init container filed,
+	//       but duplicated name error can occurs if the dst namespace has same named runtime.
+	case common.AlluxioRuntime:
+		configMapName := mountedRuntimeName + "-config"
+		err := copyConfigMap(client, configMapName, mountedRuntimeNamespace, configMapName, refNameSpace, ownerReference)
+		if err != nil {
+			return err
+		}
+	case common.JuiceFSRuntime:
+		configMapName := mountedRuntimeName + "-config"
+		err := copyConfigMap(client, configMapName, mountedRuntimeNamespace, configMapName, refNameSpace, ownerReference)
+		if err != nil {
+			return err
+		}
+	case common.GooseFSRuntime:
+		configMapName := mountedRuntimeName + "-config"
+		err := copyConfigMap(client, configMapName, mountedRuntimeNamespace, configMapName, refNameSpace, ownerReference)
+		if err != nil {
+			return err
+		}
+	case common.JindoRuntime:
+		clientConfigMapName := mountedRuntimeName + "-jindofs-client-config"
+		err := copyConfigMap(client, clientConfigMapName, mountedRuntimeNamespace, clientConfigMapName, refNameSpace, ownerReference)
+		if err != nil {
+			return err
+		}
+		configMapName := mountedRuntimeName + "-jindofs-config"
+		err = copyConfigMap(client, configMapName, mountedRuntimeNamespace, configMapName, refNameSpace, ownerReference)
+		if err != nil {
+			return err
+		}
+	case common.ThinRuntime:
+		runtimesetConfigMapName := mountedRuntimeName + "-runtimeset"
+		err := copyConfigMap(client, runtimesetConfigMapName, mountedRuntimeNamespace, runtimesetConfigMapName, refNameSpace, ownerReference)
+		if err != nil {
+			return err
+		}
+	default:
+		err := fmt.Errorf("fail to get configmap for runtime type: %s", mountedRuntimeType)
+		return err
+	}
+
+	return nil
+}
+
+func copyConfigMap(client client.Client, srcName string, srcNameSpace string, dstName string, dstNameSpace string, reference metav1.OwnerReference) error {
```

Comment:
You can use `types.NamespacedName` instead of `name` and `namespace` for short

Reply:
fixed.